src/form.ts

```diff
@@ -0,0 +1,162 @@
+    results: Results
+
+    constructor(fields: Fields) {
+        this.fields = fields
+    }
+
+    chooseFields(t: Terminal) {
+        const enabled = new Array(this.fields.length).fill(false)
+        let current = 0
+        return new Promise<Array<boolean>>((resolve, reject) => {
+            t.write("\n  Choose fields to edit:")
+            t.write("\n  (Press Enter to select/deselect, D when done, Escape to cancel)")
+            t.write("\n  " + this.fields.map(f => `[ ] ${f.prompt}: ${f.default}`).join('\n  '))
+            t.write("\x1B[4;4H") // move cursor to first field
+            const disposable = t.onKey(ev => {
+                const key = ev.domEvent.key
+                const char = !enabled[current] ? 'X' : ' '
+                switch (key) {
+                    case "Escape":
+                        reject()
+                        return
+                    case "ArrowUp":
+                        if (current > 0) {
+                            current--
+                            t.write("\x1B[A")
+                        }
+                        break
+                    case "ArrowDown":
+                        if (current < enabled.length - 1) {
+                            current++
+                            t.write("\x1B[B")
+                        }
+                        break
+                    case "Enter":
+                        enabled[current] = !enabled[current]
+                        t.write(char + "\x1B[1D")
+                        return
+                    case 'd':
+                        this.fields = this.fields.filter((_, i) => enabled[i])
+                        t.reset()
+                        disposable.dispose()
+                        resolve(enabled)
+                        return
+                }
+            })
+        })
+    }
+
+
+    start(t: Terminal) : Promise<Results> { 
+        this.i = 0
+        this.field = ''
+        this.results = []
+        return new Promise((resolve, reject) => {
+            this.writeCurrentField(t)
+            setTimeout(() => t.focus(), 0)
+            t.onKey(ev => {
+                const key = ev.domEvent.key
+                switch (key) {
+                    case "Escape":
+                        reject()
+                        return
+                    case "Backspace":
+                        if (this.field.length > 0) {
+                            t.write("\b \b")
+                            this.field = this.field.slice(0, -1)
+                        }
+                        break
+                    case "Enter":
+                        if (!this.next(t)) {
+                            resolve(this.results)
+                            return
+                        }
+                        break
+                    default:
+                        this.field += key
+                        t.write(key)
+                }
+            })
+        })
+    }
+
+    next(t: Terminal) {
```

Comment:
This function is doing more than moving the index to the next field. Please dd a function comment explaining what the function does.